### PR TITLE
Add ClusterConfig

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: cpp/log/consistent_store.h [hunk 1]
```diff
@@ -130,7 +133,17 @@ class ConsistentStore {
...
+  virtual void WatchClusterConfig(const ClusterConfigCallback& cb,
+                                  util::Task* task) = 0;
+
+  virtual util::Status SetClusterConfig(const ct::ClusterConfig& config) = 0;
+
  private:
+  std::mutex watcher_mutex_;
+  std::vector<ServingSTHCallback> sth_watchers_;
+  std::vector<ClusterNodeStateCallback> cluster_node_watchers_;
+  std::vector<ClusterConfigCallback> cluster_config_watchers_;
```
Comment (pphaneuf) [inline cpp/log/consistent_store.h]: This doesn't look right to me... Zombie code?

File: cpp/log/cluster_state_controller.h [hunk 2]
```diff
@@ -58,19 +56,24 @@ class ClusterStateController {
...
 
+  // Entry point for the config watcher callback.
+  // Called whenever the ClusterConfig is changed.
+  void OnClusterConfigUpdated(const Update<ct::ClusterConfig>& update);
+
   // Calculates the STH which should be served by the cluster, given the
   // current state of the nodes.
-  // Returns true iff there currently is a suitable STH.
-  bool CalculateServingSTH(const std::unique_lock<std::mutex>& lock);
+  // If this node is the cluster master - then the calculated servingSTH is
```
Comment (pphaneuf) [inline cpp/log/cluster_state_controller.h]: Missing space?

File: cpp/log/consistent_store.h (hunk 1 above)
Comment (AlCutter) [inline cpp/log/consistent_store.h]: Yup, thanks, dinged.

File: cpp/log/cluster_state_controller.h (hunk 2 above)
Comment (AlCutter) [inline cpp/log/cluster_state_controller.h]: added

**Comment by pphaneuf** [comment]
LGTM, pending Travis.

**Pull request merged**